Sources/HTTP/Message/HTTPMessage.swift

```diff
@@ -71,6 +71,19 @@ public struct HTTPOnUpgrade: Codable {
     }
 }
 
+extension HTTPMessage {
+    /// Sets Content-Length / Transfer-Encoding headers.
+    internal mutating func updateBodyHeaders() {
+        if let count = body.count {
+            if count > 0 {
+                headers[.contentLength] = count.description
```

Comment:
See #212

Reply:
That one is tough because currently the appendValue/removeValue API is kept private (which is great if possible because it minimizes risk of misuse).